source/solvers/vof_algebraic_interface_reinitialization.cc

```diff
@@ -0,0 +1,827 @@
+
+  DoFTools::make_hanging_node_constraints(this->dof_handler,
+                                          this->zero_constraints);
+
+  for (auto const &[id, type] :
+       this->simulation_parameters.boundary_conditions.type)
+    {
+      if (type == BoundaryConditions::BoundaryType::periodic)
+        {
+          DoFTools::make_periodicity_constraints(
+            this->dof_handler,
+            id,
+            this->simulation_parameters.boundary_conditions.periodic_neighbor_id
+              .at(id),
+            this->simulation_parameters.boundary_conditions.periodic_direction
+              .at(id),
+            this->zero_constraints);
+        }
+    }
+  for (auto const &[id, type] :
+       this->simulation_parameters.boundary_conditions_vof.type)
+    {
+      if (type == BoundaryConditions::BoundaryType::vof_dirichlet)
+        {
+          VectorTools::interpolate_boundary_values(
+            this->dof_handler,
+            id,
+            Functions::ZeroFunction<dim>(),
+            this->zero_constraints);
+        }
+    }
+  this->zero_constraints.close();
+}
+
+
+template <int dim>
+void
+VOFAlgebraicInterfaceReinitialization<dim>::define_non_zero_constraints()
+{
+  this->nonzero_constraints.clear();
+  this->nonzero_constraints.reinit(this->locally_relevant_dofs);
+
+  DoFTools::make_hanging_node_constraints(this->dof_handler,
+                                          this->nonzero_constraints);
+
+  for (auto const &[id, type] :
+       this->simulation_parameters.boundary_conditions.type)
+    {
+      if (type == BoundaryConditions::BoundaryType::periodic)
+        {
+          DoFTools::make_periodicity_constraints(
+            this->dof_handler,
+            id,
+            this->simulation_parameters.boundary_conditions.periodic_neighbor_id
+              .at(id),
+            this->simulation_parameters.boundary_conditions.periodic_direction
+              .at(id),
+            this->nonzero_constraints);
+        }
+    }
+  for (auto const &[id, type] :
+       this->simulation_parameters.boundary_conditions_vof.type)
+    {
+      if (type == BoundaryConditions::BoundaryType::vof_dirichlet)
+        {
+          VectorTools::interpolate_boundary_values(
+            this->dof_handler,
+            id,
+            *this->simulation_parameters.boundary_conditions_vof.phase_fraction
+               .at(id),
+            this->nonzero_constraints);
+        }
+    }
+  this->nonzero_constraints.close();
+}
+
+
+template <int dim>
+void
+VOFAlgebraicInterfaceReinitialization<dim>::set_initial_conditions()
+{
+  // Get VOF DoFHandler
+  const DoFHandler<dim> *dof_handler_vof =
+    this->multiphysics_interface->get_dof_handler(PhysicsID::VOF);
+
+  // Interpolate VOF solution to algebraic interface reinitialization
+  VectorTools::interpolate_to_different_mesh(
+    *dof_handler_vof,
+    *this->multiphysics_interface->get_solution(PhysicsID::VOF),
+    this->dof_handler,
+    this->newton_update);
```

Comment:
It's funny that you need to "interpolate to different mesh" because in the end it's the same triangulation no?